File: tests/codegraph/fixtures/sample.py
"""Fixture module for codegraph Python extractor tests."""

from __future__ import annotations

import os
from pathlib import Path as P


def top_level_helper(x: int) -> int:
    return x + 1


class Animal:
    def speak(self) -> str:
        return "generic sound"


class Dog(Animal):
    def speak(self) -> str:
        os.getenv("IGNORED")
        return self._bark_style()

    def _bark_style(self) -> str:
        return "woof"


def run() -> None:
    d = Dog()
    d.speak()
    top_level_helper(1)
    P(".").resolve()

File: tests/codegraph/fixtures/sample.ts
// Fixture module for codegraph TypeScript extractor tests.

import { helper } from './utils'
import type { Config } from './config'

export interface Pettable {
  pet(): void
}

interface Namable extends Pettable {
  name: string
}

export abstract class Animal {
  protected noises: number = 0
  abstract speak(): string
}

export class Dog extends Animal implements Namable {
  constructor(public name: string) {
    super()
  }

  speak(): string {
    return this.barkStyle()
  }

  private barkStyle(): string {
    helper()
    return 'woof'
  }

  pet(): void {
    this.noises += 1
  }
}

export const run = (): void => {
  const d = new Dog('rex')
  d.speak()
  helper()
}

export const run2 = function () {
  run()
}

export enum Kind {
  Good,
  Bad,
}
